snooty/parser.py

```diff
@@ -654,16 +654,33 @@ def handle_directive(
         elif name == "openapi":
             # Parsing should be done by the OpenAPI renderer on the frontend by default
             uses_rst = options.get("uses-rst", False)
+            # The frontend will grab the file contents from Realm, using the argument as its key
+            uses_realm = options.get("uses-realm", False)
 
             if argument_text is None:
+                if uses_realm or uses_rst:
+                    self.diagnostics.append(ExpectedPathArg(name, line))
+                    return doc
+
                 # Check if argument is a url instead
                 url_argument = None
+                spec = None
                 try:
                     url_argument = argument[0].refuri
+                    with urllib.request.urlopen(url_argument) as url:
```

Comment:
Two issues with this:

1) urllib is actually dispreferred despite being in the stdlib: if you need to make a direct HTTP request, better to use, well, `requests` (which uses `urllib3` behind the scenes)
2) This request will be made every time the page is parsed, which will happen as the writer types. This is really problematic! It would be nice to have a proper mechanism for dealing with this kind of thing, but for now, probably better to use our `util.HTTPCache` class to at least make sure that the language server doesn't have to keep re-fetching the file every ~100ms. See: https://github.com/mongodb/snooty-parser/blob/master/snooty/intersphinx.py#L149